### PR TITLE
Add cpp solution to usaco-736.mdx

### PR DESCRIPTION
**Pull request by anirudhc1229**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as
done._

- [x] I have tested my code
- [x] I have followed the code style guidelines mentioned
      [here](https://usaco.guide/general/adding-solution/#code-conventions)
- [x] I have properly formatted the files according to the steps mentioned
      [here](https://usaco.guide/general/adding-solution#steps)

### CODE REVIEW

File: solutions/usaco-736.mdx [hunk 1]
```diff
@@ -126,6 +126,72 @@ file_out.close()
...
+    ofstream fout("cownomics.out");
+
+    int N, M;
+    fin >> N >> M;
+    vector<string> spotty(N);
+    for (int i = 0; i < N; i++) fin >> spotty[i];
+    vector<string> plain(N);
+    for (int i = 0; i < N; i++) fin >> plain[i];
+
+    map<char, int> spottyCt;
```
Comment (bqi343) [inline solutions/usaco-736.mdx:158]: why not have these maps inside the loop?

File: solutions/usaco-736.mdx [hunk 2]
```diff
@@ -126,6 +126,72 @@ file_out.close()
...
+    map<char, int> plainCt;
+    int ans = 0;
+    
+    for (int i = 0; i < M; i++) {
+
```
Comment (bqi343) [inline solutions/usaco-736.mdx:163]: there's really no need for all these blank lines

https://google.github.io/styleguide/cppguide.html#Vertical_Whitespace

**Review by bqi343 — CHANGES_REQUESTED** [review]
mostly good, but

> I have followed the code style guidelines mentioned here

nope

File: solutions/usaco-736.mdx [hunk 3]
```diff
@@ -126,6 +126,65 @@ file_out.close()
...
+    vector<string> spotty(N);
+    for (int i = 0; i < N; i++) fin >> spotty[i];
+    vector<string> plain(N);
+    for (int i = 0; i < N; i++) fin >> plain[i];
+    int ans = 0;
+    
+    for (int i = 0; i < M; i++) {
+        // initialize letter counters for spotty and plain strings
+	map<char, int> spottyCt, plainCt;
+	spottyCt['A'] = spottyCt['C'] = spottyCt['G'] = spottyCt['T'] = 0;
```
Comment (bqi343) [inline solutions/usaco-736.mdx:162]: these lines are unnecessary

**Review by bqi343 — CHANGES_REQUESTED** [review]
https://github.com/cpinitiative/usaco-guide/pull/2479#discussion_r779998042 also applies here

File: solutions/usaco-736.mdx [hunk 4]
```diff
@@ -126,6 +126,65 @@ file_out.close()
 ## Solution 2 - Hashset
```
Comment (bqi343) [inline solutions/usaco-736.mdx:126]:
```suggestion
## Solution 2 - Set
```

`std::map` isn't a hashset.

**Review by bqi343 — CHANGES_REQUESTED** [review]

File: solutions/usaco-736.mdx (hunk 3 above)
Comment (bqi343) [reply solutions/usaco-736.mdx:162]: why was this marked as resolved?

File: solutions/usaco-736.mdx [hunk 5]
```diff
@@ -123,9 +123,68 @@ file_out.close()
...
+#include <fstream>
+#include <string>
+#include <vector>
+#include <algorithm>
+#include <map>
+
+using namespace std;
+
+int main() {
+
```
Comment (bqi343) [inline solutions/usaco-736.mdx:147]: not sure why this empty newline wasn't removed

File: solutions/usaco-736.mdx [hunk 6]
```diff
@@ -126,6 +126,65 @@ file_out.close()
...
+    vector<string> plain(N);
+    for (int i = 0; i < N; i++) fin >> plain[i];
+    int ans = 0;
+    
+    for (int i = 0; i < M; i++) {
+        // initialize letter counters for spotty and plain strings
+	map<char, int> spottyCt, plainCt;
+	spottyCt['A'] = spottyCt['C'] = spottyCt['G'] = spottyCt['T'] = 0;
+        plainCt['A'] = plainCt['C'] = plainCt['G'] = plainCt['T'] = 0;
+	// count occurrences of each letter in each string at index M
```
[the remark this answers is not in the document]
Comment (bqi343) [reply solutions/usaco-736.mdx:164]: same, this isn't resolved

**Comment by SansPapyrus683** [comment]
if you want i can just format the pr for you

**Pull request closed without merging**